### PR TITLE
Key sheet input update

### PR DESCRIPTION
**Pull request by BigRoma**
Add button and extend component props.

### CODE REVIEW

File: kit/inputs/src/UIKeyTextView.tsx [hunk 1]
```diff
@@ -93,6 +103,16 @@ export function useKeyTextView(
         };
     }, [inputHasValue, hasInvalidChars, hasProperLength, isFocused]);
 
+    React.useEffect(() => {
```
Comment (savelichalex) [inline kit/inputs/src/UIKeyTextView.tsx:100]: What do you think to split to two different `useEffect`s, I'm not quite understood why you want `onError` to be called, when only `success` is changed

File: stories/browser/src/UIKeySheet.tsx [hunk 2]
```diff
@@ -2,19 +2,31 @@ import * as React from 'react';
...
 
 import { UIKeyTextView, UIMaterialTextViewRef } from '@tonlabs/uikit.inputs';
+import { UIBoxButton } from '@tonlabs/uikit.controls';
 
 import { UIPullerSheet } from './UIPullerSheet';
 
+function useKeySheetContent() {
+    const [error, setError] = React.useState(false);
+    const [success, setSuccess] = React.useState(false);
+    const [key, setKey] = React.useState('');
```
Comment (savelichalex) [inline stories/browser/src/UIKeySheet.tsx]: You can use `useRef` here instead, as the change of it doesn't update any UI, so there is no need to trigger re-render here

File: kit/inputs/src/UIKeyTextView.tsx [hunk 3]
```diff
@@ -13,12 +13,18 @@ import {
...
+type OnError = (error: boolean) => void | Promise<void>;
 
 export function useKeyTextView(
     ref: React.Ref<TextInput> | null,
     isFocused: boolean,
     props: UIMaterialTextViewProps & {
         onDone: OnDone;
+        onSuccess?: OnSuccess;
+        onError?: OnError;
+        onChangeKey?: onChange;
```
Comment (savelichalex) [inline kit/inputs/src/UIKeyTextView.tsx]: You could've just pass `onChangeText`, it should be done automatically, without any additional code
https://github.com/tonlabs/UIKit/blob/7edd865e1179ff18d6cdc64ede5f465d4f1af23d/kit/inputs/src/UITextView.tsx#L132
https://github.com/tonlabs/UIKit/blob/7edd865e1179ff18d6cdc64ede5f465d4f1af23d/kit/inputs/src/UITextView.tsx#L184

File: kit/inputs/src/UIKeyTextView.tsx [hunk 4]
```diff
@@ -47,6 +53,10 @@ export function useKeyTextView(
             if (hasInvalidChars !== isCharsInvalid) {
                 setHasInvalidChars(isCharsInvalid);
             }
+
+            if (onChangeKey) {
```
Comment (savelichalex) [inline kit/inputs/src/UIKeyTextView.tsx]: Same as above, using `onChangeText` you may not write this additional code

File: localization/src/languages/de.json [hunk 5]
```diff
@@ -89,7 +90,8 @@
             "PickCipher": "Pick from encryption keys",
             "CipherKeys": "Encryption keys",
             "AddCipherKey": "Add an encryption key",
-            "CipherKeyLabel": "Encryption key"
+            "CipherKeyLabel": "Encryption key",
+            "Encrypt": "Encrypt"
```
Comment (sertony) [inline localization/src/languages/de.json]: It's not necessary `encrypt`, as the could also be used for decryption....

File: stories/browser/src/UIKeySheet.tsx [hunk 6]
```diff
@@ -30,9 +42,21 @@ function UIKeySheetContent({
...
-            <UIKeyTextView ref={keyRef} label={label} onDone={onKeyRetrieved} />
+            <UIKeyTextView
+                ref={keyRef}
+                label={label}
+                onDone={onKeyRetrieved}
+                onSuccess={setSuccess}
+                onError={setError}
+                onChangeKey={setKey}
+            />
+            <UIBoxButton onPress={onSend} disabled={!success || error} title={buttonTitle} />
```
Comment (sertony) [inline stories/browser/src/UIKeySheet.tsx]: `!success || error` result could be memoized

File: kit/inputs/src/UIKeyTextView.tsx (hunk 3 above)
Comment (BigRoma) [reply kit/inputs/src/UIKeyTextView.tsx]: Yep, it works, but this prop excludes from the original component.

**Review by sertony — COMMENTED** [review]
Also, AFAIR this component has some broken logic regarding `enter` press...

For instance if you press "Enter" when key is not fully "entered" in the input, it will return it to to browser (which is bad).
This was probably fixed in this PR: https://github.com/tonlabs/UIKit/pull/1344/files#diff-2fd447804e295951c1314a3b3b4964d74093e7b67d69b5da652fd67531ea1cf7R89

Need to check it and apply this change (or alike).

File: kit/inputs/src/UIKeyTextView.tsx (hunk 3 above)
Comment (savelichalex) [reply kit/inputs/src/UIKeyTextView.tsx]: Yeah, probably need to do sth with types to allow to pass it there :)

File: kit/inputs/src/UIKeyTextView.tsx (hunk 1 above)
Comment (BigRoma) [reply kit/inputs/src/UIKeyTextView.tsx:100]: https://github.com/tonlabs/UIKit/pull/1476/commits/9a0447c5cc35417b0ecb3622a1aaebf571c8e06b

File: stories/browser/src/UIKeySheet.tsx (hunk 2 above)
Comment (BigRoma) [reply stories/browser/src/UIKeySheet.tsx]: https://github.com/tonlabs/UIKit/pull/1476/commits/9a0447c5cc35417b0ecb3622a1aaebf571c8e06b

File: kit/inputs/src/UIKeyTextView.tsx (hunk 3 above)
Comment (BigRoma) [reply kit/inputs/src/UIKeyTextView.tsx]: https://github.com/tonlabs/UIKit/pull/1476/commits/9a0447c5cc35417b0ecb3622a1aaebf571c8e06b

File: kit/inputs/src/UIKeyTextView.tsx (hunk 4 above)
Comment (BigRoma) [reply kit/inputs/src/UIKeyTextView.tsx]: https://github.com/tonlabs/UIKit/pull/1476/commits/9a0447c5cc35417b0ecb3622a1aaebf571c8e06b

File: localization/src/languages/de.json (hunk 5 above)
Comment (BigRoma) [reply localization/src/languages/de.json]: https://github.com/tonlabs/UIKit/pull/1476/commits/9a0447c5cc35417b0ecb3622a1aaebf571c8e06b

File: stories/browser/src/UIKeySheet.tsx (hunk 6 above)
Comment (BigRoma) [reply stories/browser/src/UIKeySheet.tsx]: https://github.com/tonlabs/UIKit/pull/1476/commits/c1c400bc4e4e15c9d33964fd64633a8ca03fedd4

File: localization/src/languages/ko.json [hunk 7]
```diff
@@ -53,7 +53,8 @@
             "CipherKeys": "암호화 키",
             "EnterKey": "수동으로 암호화 키 입력",
             "PickCipher": "암호화 키에서 선택",
-            "Prompt": "암호화 키 선택:"
+            "Prompt": "암호화 키 선택:",
+            "UseKey": "Use key"
```
Comment (sertony) [inline localization/src/languages/ko.json]: Can you ask @kkkhesha to translate, please?

File: localization/src/languages/ko.json [hunk 8]
```diff
@@ -66,7 +67,8 @@
             "Prompt": "어떻게 서명하시겠습니까?",
             "SecurityCards": "보안 카드",
             "Signatures": "서명",
-            "UseSecurityCard": "보안 카드 사용"
+            "UseSecurityCard": "보안 카드 사용",
+            "Sign": "Sign"
```
Comment (sertony) [inline localization/src/languages/ko.json]: Same

**Review by sertony — APPROVED** [review]

File: stories/browser/src/UIKeySheet.tsx [hunk 9]
```diff
@@ -2,19 +2,35 @@ import * as React from 'react';
...
 import { UIPullerSheet } from './UIPullerSheet';
 
+function useKeySheetContent() {
+    const [error, setError] = React.useState(false);
+    const [success, setSuccess] = React.useState(false);
+
+    const key = React.useRef('');
+    const setKey = React.useCallback((val: string) => {
+        key.current = val;
+    }, []);
```
Comment (savelichalex) [inline stories/browser/src/UIKeySheet.tsx:16]: Here we go, nice! :)

**Review by savelichalex — APPROVED** [review]

File: localization/src/languages/ko.json (hunk 7 above)
Comment (kkkhesha) [reply localization/src/languages/ko.json]:
```suggestion
            "UseKey": "키 이용"
```

Comment (kkkhesha) [reply localization/src/languages/ko.json]: @sertony

File: localization/src/languages/ko.json (hunk 8 above)
Comment (kkkhesha) [reply localization/src/languages/ko.json]:
```suggestion
            "Sign": "서명"
```

**Pull request merged**